eval.go

```diff
@@ -134,13 +135,17 @@ func (o *SearchOptions) SetDefaults() {
 }
 
 func (d *indexData) Search(ctx context.Context, q query.Q, opts *SearchOptions) (sr *SearchResult, err error) {
+	var res SearchResult
```

Comment:
For other tenant-aware services, we'd tried to design it so there's a "choke point" that clearly enforces tenancy. Often this is single interface and file guarded by `//🚨 SECURITY:` comments. Right now, we are only enforcing tenancy on `Search`, but not on `List`. And there are other places where we access index data, for example to get metadata or ngram stats.

I wonder if we could push this down further to where index data is loaded/ read. Like `reader.readTOC` and `reader.readIndexData`? That would help establish a single place on the read path where tenancy is enforced, as close to the data access as possible.

Reply:
Good point. Now we wrap `indexdata` (which is a Searcher) in a `tenantAwareSearcher` that handles access. We use a similar pattern in Sourcegraph. WDYT?